src/library/dlgtagfetcher.cpp

```diff
@@ -399,4 +514,145 @@ void DlgTagFetcher::tagSelected() {
     }
     const int tagIndex = tags->currentItem()->data(0, Qt::UserRole).toInt();
     m_data.m_selectedTag = tagIndex;
+
+    if (!m_fetchedCoverArtByteArrays.isNull()) {
+        m_fetchedCoverArtByteArrays.clear();
+        m_pWFetchedCoverArtLabel->loadData(m_fetchedCoverArtByteArrays);
+        m_pWFetchedCoverArtLabel->setCoverArt(CoverInfo{},
+                QPixmap(CoverArtUtils::defaultCoverLocation()));
+    }
```

Comment:
This seems to assume that if `m_fetchedCoverArtByteArrays` is null `m_pWFetchedCoverArtLabel` was already cleared (default cover image) and wants to avoid unnecessary clearing?
```suggestion
    m_fetchedCoverArtByteArrays.clear();
    m_pWFetchedCoverArtLabel->loadData(QByteArray());
    m_pWFetchedCoverArtLabel->setCoverArt(CoverInfo{},
    QPixmap(CoverArtUtils::defaultCoverLocation()));
```

Reply:
Yeah that was the purpose. But with the suggested changes it looks better